meet_balls_app/views.py

```diff
@@ -1,5 +1,32 @@
 from django.shortcuts import render
+from django.contrib.auth import authenticate
+from django.contrib import messages
 
 
 def home(request):
     return render(request, 'meet_balls_app/home.html')
+
+
+def login(request):
+    if request.method == 'POST':
+        username = request.POST.get('username')
+        password = request.POST.get('password')
+        user = authenticate(request, username=username, password=password)
+        if user is not None:
+            # login(request, user)
+            return render(request, 'meet_balls_app/home.html',
+                          {'user': {'is_authenticated': True, 'username': username}})
+        else:
+            error_message = "Invalid username or password."
+            messages.error(request, error_message)
+            return render(request, 'meet_balls_app/login.html', {'user': {'is_authenticated': False, 'username': None}})
+    else:
+        return render(request, 'meet_balls_app/login.html', {'user': {'is_authenticated': False, 'username': None}})
+
+
+def logout(request):
+    return render(request, 'meet_balls_app/home.html', {'user': {'is_authenticated': False}})
+
+
+def register(request):
```

Comment:
Are we testing this one?

Reply:
Hey @yftacherzog , on this matter, @IdanZimi  and I are not testing this because we didn't implement registration logic yet. We have the registater link on our html, in this section : `<p>Don't have an account? <a href="/register">Register now</a>.</p>`, and therefore the propriate path in the urls.py file `path('register/', app_views.register, name='register'),`
that is why we put this section with testing, because we think there is nothing to test on this matter currently, and we'll add it once will implement registration mechanism. 
On the other hand, we can remove the a href so this part in the HTML won't lead to anywhere, and we could also remove the path from the urls.py file, and this line of code here in the views.py file.

WDYT @yftacherzog @Yuval-Vino

Reply:
Either you implement a test verifying the current logic (e.g. verify the redirection), or you remove the function.

Reply:
@yftacherzog we removed the function.